### PR TITLE
Improve section parsing  (don't include the last line)

### PR DESCRIPTION
**Pull request by wilzbach**
I ran into quite some trouble when trying to turn all Dscanner checks for Phobos.

I hope the added examples are self-documentary.

### CODE REVIEW

File: src/ddoc/macros.d [hunk 1]
```diff
@@ -502,8 +502,11 @@ private bool parseAsKeyValuePair(ref Lexer olexer, ref string key, ref string va
 	}
 	else
 		return false;
-	while (!lexer.empty && lexer.front.type == Type.whitespace)
+
+	// parentheses might be used in the key
+	while (!lexer.empty && lexer.front.type.among(Type.whitespace, Type.lParen, Type.rParen, Type.word))
```
Comment (wilzbach) [inline src/ddoc/macros.d]: One also might argue that the occurrence at Phobos should be fixed:

https://github.com/dlang/phobos/blob/548630d2a9ba95f079193d7b47ed8cd685c0f7c8/std/regex/package.d#L353

Comment (wilzbach) [inline src/ddoc/macros.d]: -> https://github.com/dlang/phobos/pull/5499

Comment (Hackerpilot) [inline src/ddoc/macros.d]: Having looked at the changes in Phobos it seems that Phobos was wrong...

Comment (Hackerpilot) [inline src/ddoc/macros.d]: Ddoc spec says:
> Each line that starts with an identifier followed by an '=' starts a new parameter description.

I'm pretty sure that parens aren't part of identifiers anywhere else in D, so is it really a good idea to make an exception here?

**Comment by wilzbach** [comment]
> Having looked at the changes in Phobos it seems that Phobos was wrong...

Luckily my PR at Phobos was merged, so I could drop the {L,R}parens allowance in Ddoc parameters :)

**Pull request merged**